### PR TITLE
Bar Chart Animations with flag

### PR DESCRIPTION
**Pull request by Golodhros**
Adding the missing animations on bar charts, both horizontal and vertical

## Description
I am adding animations to the bar charts and improving the demo, as well as adjusting the line chart demo spacing and load to fit better on mobile sizes.

## Motivation and Context
Related to #243, I think we should add animations on the bar chart for the sake of completion.

## How Has This Been Tested?
Ran tests and checked on demos

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: demos/demo-line.js [hunk 1]
```diff
@@ -10,12 +10,14 @@ var _ = require('underscore'),
     line = require('./../src/charts/line'),
     tooltip = require('./../src/charts/tooltip'),
     dataBuilder = require('./../test/fixtures/lineChartDataBuilder'),
-    colorSelectorHelper = require('./helpers/colorSelector');
+    colorSelectorHelper = require('./helpers/colorSelector'),
+
+    barMargin = {top:60, bottom: 50, left: 50, right: 30};
```
Comment (rwholey-eb) [inline demos/demo-line.js]: barMargin, this is line demo?

File: src/charts/bar.js [hunk 2]
```diff
@@ -317,6 +328,81 @@ define(function(require) {
         }
 
         /**
+         * Draws and animates the bars along the x axis
+         * @param  {D3Selection} bars Selection of bars
+         * @return {void}
+         */
+        function drawAnimatedHorizontalBars(bars) {
+            // Enter + Update
```
Comment (rwholey-eb) [inline src/charts/bar.js]: what about trying to combine some of the logic here? I see this divergent pattern as a potential problem in the future, where horizontal and vertical end up with different behaviors/features

File: demos/demo-line.js (hunk 1 above)
Comment (Golodhros) [inline demos/demo-line.js]: Oops!! Good call!

File: src/charts/bar.js (hunk 2 above)
Comment (Golodhros) [inline src/charts/bar.js]: Yeah, totally. I am not super happy with this approach either.
I will think about how to do it other way...

**Pull request merged**